hkdf/src/lib.rs

```diff
@@ -167,62 +142,28 @@ where
 
     /// Completes the HKDF-Extract operation, returning both the generated pseudorandom key and
     /// `Hkdf` struct for expanding.
-    pub fn finalize(self) -> (Output<D::Core>, Hkdf<D>) {
+    pub fn finalize(self) -> (Output<SimpleHmac<D>>, Hkdf<D>) {
         let prk = self.hmac.finalize_fixed();
         let hkdf = Hkdf::from_prk(&prk).expect("PRK size is correct");
         (prk, hkdf)
     }
 }
 
-impl<D> fmt::Debug for HkdfExtract<D>
-where
-    D: CoreProxy,
-    D::Core: HashMarker
-        + AlgorithmName
-        + UpdateCore
-        + FixedOutputCore
-        + BufferKindUser<BufferKind = Eager>
-        + Default
-        + Clone,
-    <D::Core as BlockSizeUser>::BlockSize: IsLess<U256>,
-    Le<<D::Core as BlockSizeUser>::BlockSize, U256>: NonZero,
-{
+impl<D: Digest + BlockSizeUser + fmt::Debug> fmt::Debug for HkdfExtract<D> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.write_str("HkdfExtract<")?;
-        <D::Core as AlgorithmName>::write_alg_name(f)?;
+        D::fmt(&D::new(), f)?;
```

Comment:
Is there a better way to do this?

Reply:
I'd probably replace the whole thing with a `write!` directive or use `DebugStruct`:

```suggestion
        write!(f, "HkdfExtract<{:?}> { ... } ", &D::new())
```